src/main/java/io/pravega/connectors/flink/FlinkPravegaReader.java

```diff
@@ -220,6 +258,150 @@ private void triggerCheckpoint(String checkpointIdentifier) throws FlinkExceptio
+            return readerGroup.getOnlineReaders().stream().collect(Collectors.joining(separator));
+        }
+    }
+
+    /**
+     * Gauge for getting stream name information from reader group.
+     */
+    private static class StreamNamesGauge implements Gauge<String> {
+
+        private final ReaderGroup readerGroup;
+        private final String separator = ",";
+
+        public StreamNamesGauge(ReaderGroup readerGroup) {
+            this.readerGroup = readerGroup;
+        }
+
+        @Override
+        public String getValue() {
+            return readerGroup.getStreamNames().stream().collect(Collectors.joining(separator));
+        }
+    }
+
+    /**
+     * Gauge for getting position information of each segment of a stream from reader group.
+     */
+    private static class SegmentPositionsGauge implements Gauge<String> {
+
+        private final ReaderGroup readerGroup;
+        private final String stream;
+
+        public SegmentPositionsGauge(ReaderGroup readerGroup, String stream) {
+            this.readerGroup = readerGroup;
+            this.stream = stream;
+        }
+
+        @Override
+        public String getValue() {
+            StringBuilder builder = new StringBuilder();
+            builder.append("stream=").append(stream).append(", segments={");
+            Map<Stream, StreamCut> streamCuts = readerGroup.getStreamCuts();
+            Optional<Map.Entry<Stream, StreamCut>> optionalStreamCutEntry =
+                    streamCuts.entrySet().stream().filter(e -> e.getKey().getStreamName().equals(stream)).findFirst();
+            if (optionalStreamCutEntry.isPresent()) {
+                optionalStreamCutEntry.get().getValue().asImpl().getPositions().entrySet().stream().forEach(entry -> {
+                    builder.append(entry.getKey()).append("=").append(entry.getValue()).append(", ");
+                });
+            }
+            builder.append("}");
+            return builder.toString();
+        }
+    }
+
+    /**
+     * register reader group metrics
+     *
+     */
+    private void registerMetrics() {
+        ReaderGroup readerGroup = createReaderGroup();
+        MetricGroup pravegaReaderMetricGroup = getRuntimeContext().getMetricGroup().addGroup(PRAVEGA_READER_METRICS_GROUP);
+        MetricGroup readerGroupMetricGroup = pravegaReaderMetricGroup.addGroup(READER_GROUP_METRICS_GROUP);
+        readerGroupMetricGroup.gauge(UNREAD_BYTES_METRICS_GAUGE, new UnreadBytesGauge(readerGroup));
+        readerGroupMetricGroup.gauge(READER_GROUP_NAME_METRICS_GAUGE, new ReaderGroupNameGauge(readerGroup));
+        readerGroupMetricGroup.gauge(SCOPE_NAME_METRICS_GAUGE, new ScopeNameGauge(readerGroup));
+        readerGroupMetricGroup.gauge(ONLINE_READERS_METRICS_GAUGE, new OnlineReadersGauge(readerGroup));
+        readerGroupMetricGroup.gauge(STREAM_NAMES_METRICS_GAUGE, new StreamNamesGauge(readerGroup));
+
+        Map<Stream, StreamCut> streamCuts = readerGroup.getStreamCuts();
```

Comment:
`getStreamCuts()` is used only to fetch the scope and stream information to create appropriate metric groups since the `getStreamNames()` API does not return fully qualified stream details (https://github.com/pravega/pravega/issues/2860).